### PR TITLE
Refactor common code from google.oauth2.flow to google.oauth2.oauthlib

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: docs/reference/google.oauth2.oauthlib.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+google.oauth2.oauthlib module
+=============================
+
+.. automodule:: google.oauth2.oauthlib
```
Comment (dhermes) [inline docs/reference/google.oauth2.oauthlib.rst]: This is new?

File: google/oauth2/oauthlib.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
+# Copyright 2016 Google Inc.
```
Comment (dhermes) [inline google/oauth2/oauthlib.py]: 2017

File: google/oauth2/oauthlib.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import requests_oauthlib
+
+import google.oauth2.credentials
+
+_REQUIRED_CONFIG_KEYS = frozenset(('auth_uri', 'token_uri', 'client_id'))
+
+
+def session_from_client_config(client_config, scopes, **kwargs):
+    """Creates a :class:`requests_oauthlib.OAuth2Session` from client
+    configuration loaded from a Google-format client secrets file.
```
Comment (dhermes) [inline google/oauth2/oauthlib.py]: :cry:

File: docs/reference/google.oauth2.oauthlib.rst (hunk 1 above)
Comment (jonparrott) [inline docs/reference/google.oauth2.oauthlib.rst]: Yes, I pulled out a bunch of the lower-level code in `flow` into `oauthlib`.

File: google/oauth2/oauthlib.py (hunk 3 above)
Comment (jonparrott) [inline google/oauth2/oauthlib.py]: Why :cry: ?

File: google/oauth2/oauthlib.py (hunk 2 above)
Comment (jonparrott) [inline google/oauth2/oauthlib.py]: Fixed.

File: google/oauth2/oauthlib.py (hunk 3 above)
Comment (jonparrott) [inline google/oauth2/oauthlib.py]: I'm gonna go ahead and merge, but if you had a concern here I'll be happy to address.

Comment (dhermes) [inline google/oauth2/oauthlib.py]: The cry face was because one-liner on two lines

Comment (jonparrott) [inline google/oauth2/oauthlib.py]: Oh, I haven't broken you on that yet? :P

Comment (dhermes) [inline google/oauth2/oauthlib.py]: Neveeeerrrrrrrrrrrr!

**Pull request merged**